sys/Makefile.dep

```diff
@@ -698,6 +698,7 @@ ifneq (,$(filter xtimer,$(USEMODULE)))
     endif
   else
   # ztimer_xtimer_compat is used, all of *xtimer's API will be mapped on ztimer.*
+  NO_AUTOLOCATE += xtimer
```

Comment:
Won't this be affected by the order of inclusion?

Reply:
The location of modules is done fully again at each recursion step. If xtimer would be auto-located in a transient state, it would be dropped later on. But some care should be taken so that an `Makefile.dep` doesn't do anything `xtimer` related when `ztimer_xtimer_compat` is also used. But since there is no `Makefile.dep` in `sys/xtimer` and in `sys/Makefile.dep` (above code) the correct guards are in place, this should just work.

A bit of a foot gun, but choosing different implementation options for the same API is just not properly modeled in the build system, so some friction is expected.